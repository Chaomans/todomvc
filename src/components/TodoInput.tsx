type TodoInputProps = {
    createTodo: (event: React.KeyboardEvent<HTMLInputElement>) => void;
    onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
};

const TodoInput = ({ createTodo, onChange }: TodoInputProps) => {
    return (
            <input
                type="text"
                placeholder="Whats's next"
                className="new-todo"
                onKeyDown={createTodo}
                onChange={onChange}
                autoFocus
            />
    );
};

export default TodoInput;
