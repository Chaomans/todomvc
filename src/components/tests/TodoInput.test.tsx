import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import TodoInput from '../TodoInput';

describe("TodoInput component", () => {
    it("Should render properly", () => {
        const { asFragment } = render(
            <TodoInput
                createTodo={() => {}}
                onChange = {() => {}}
            />
        )

        expect(asFragment()).toMatchSnapshot();
    });

    it("Should prevent parent when key pressed", async () => {
        const mockCreateTodo = jest.fn();
        const mockOnChange = jest.fn();

        render(
            <TodoInput
                createTodo={mockCreateTodo}
                onChange={mockOnChange}
            />
        )

        await userEvent.keyboard("x")
        expect(mockCreateTodo).toHaveBeenCalledTimes(1);
        expect(mockOnChange).toHaveBeenCalledTimes(1);
    })
})