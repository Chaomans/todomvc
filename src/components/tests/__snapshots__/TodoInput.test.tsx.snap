// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoInput component Should render properly 1`] = `
<DocumentFragment>
  <input
    class="new-todo"
    placeholder="Whats's next"
    type="text"
  />
</DocumentFragment>
`;
